src/util/irep.h

```diff
@@ -440,9 +441,15 @@ class irept
   long long get_long_long(const irep_namet &name) const;
 
   void set(const irep_namet &name, const irep_idt &value)
-  { add(name).id(value); }
+  {
+    add(name, irept(value));
+  }
   void set(const irep_namet &name, const irept &irep)
   { add(name, irep); }
+  void set(const irep_namet &name, irept &&irep)
+  {
+    add(name, std::move(irep));
+  }
   void set(const irep_namet &name, const long long value);
```

Comment:
Since this will copy eventually, simply have one variant, with just `irept`, and then move that?

Reply:
Done.